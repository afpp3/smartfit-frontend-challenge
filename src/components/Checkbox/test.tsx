import { screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { renderWithTheme } from 'helpers/test/helpers'

import Checkbox from '.'

describe('<Checkbox />', () => {
  it('should render with label', () => {
    renderWithTheme(<Checkbox label="label" labelFor="checkbox" />)

    expect(screen.getByRole('checkbox')).toBeInTheDocument()

    expect(screen.getByLabelText(/label/i)).toBeInTheDocument()

    expect(screen.getByText(/label/i)).toHaveAttribute('for', 'checkbox')
  })

  it('should dispatch onCheck when status changes', async () => {
    const onCheck = jest.fn()

    renderWithTheme(
      <Checkbox label="label" labelFor="checkbox" onCheck={onCheck} />
    )
    expect(onCheck).not.toHaveBeenCalled()

    userEvent.click(screen.getByRole('checkbox'))
    await waitFor(() => {
      expect(onCheck).toHaveBeenCalledTimes(1)
    })

    expect(onCheck).toHaveBeenCalledWith(true)
  })

  it('should dispatch onCheck when status changes', async () => {
    const onCheck = jest.fn()

    renderWithTheme(
      <Checkbox label="label" labelFor="checkbox" onCheck={onCheck} isChecked />
    )

    userEvent.click(screen.getByRole('checkbox'))
    await waitFor(() => {
      expect(onCheck).toHaveBeenCalledTimes(1)
    })
    expect(onCheck).toHaveBeenCalledWith(false)
  })

  it('should be acessible with tab', () => {
    renderWithTheme(<Checkbox label="label" labelFor="checkbox" />)

    expect(document.body).toHaveFocus()

    userEvent.tab()

    expect(screen.getByLabelText(/label/i)).toHaveFocus()
  })
})
