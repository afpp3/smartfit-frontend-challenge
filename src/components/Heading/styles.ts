import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    color: ${theme.colors.darkGrey};
  `}
`

export const Title = styled.h1`
  ${({ theme }) => css`
    font-weight: ${theme.font.black};
    font-size: ${theme.font.sizes.huge};
    width: min-content;
    position: relative;
    margin-bottom: ${theme.spacings.xxlarge};

    &::after {
      content: '';
      position: absolute;
      bottom: -2.6rem;
      left: 0;
      width: 11.2rem;
      border-bottom: 1rem solid ${theme.colors.darkGrey};
    }

    ${media.lessThan('small')`
      font-size: ${theme.font.sizes.xxlarge}
    `}
  `}
`

export const Description = styled.p`
  ${({ theme }) => css`
    font-weight: ${theme.font.book};
  `}
`
