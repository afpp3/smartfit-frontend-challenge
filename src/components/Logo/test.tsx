import { render, screen } from '@testing-library/react'
import Logo from '.'

describe('<Logo />', () => {
  it('should render the Logo', () => {
    const { container } = render(<Logo />)

    expect(screen.getByLabelText(/Smart Fit/i)).toBeInTheDocument()
    expect(screen.getByLabelText(/smart fit/i).parentElement).toHaveStyle({
      width: '14rem',
      height: '5.4rem'
    })
    expect(container.firstChild).toMatchSnapshot()
  })
})
