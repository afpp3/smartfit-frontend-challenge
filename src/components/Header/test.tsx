import { screen } from '@testing-library/react'
import { renderWithTheme } from 'helpers/test/helpers'

import Header from '.'

describe('<Header />', () => {
  it('should render the header with logo', () => {
    const { container } = renderWithTheme(<Header />)
    expect(screen.getByRole('img', { name: /smart fit/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
