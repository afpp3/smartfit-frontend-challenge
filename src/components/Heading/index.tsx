import * as S from './styles'

export type HeadingProps = {
  title: string
  description?: string
}

const Heading = ({ title, description }: HeadingProps) => (
  <S.Wrapper>
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
  </S.Wrapper>
)

export default Heading
