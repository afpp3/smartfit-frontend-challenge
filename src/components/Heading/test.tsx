import { screen } from '@testing-library/react'
import { renderWithTheme } from 'helpers/test/helpers'

import Heading from '.'

describe('<Heading />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Heading title="Heading" />)

    expect(
      screen.getByRole('heading', { name: /heading/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the heading with description', () => {
    renderWithTheme(<Heading title="Heading" description="Description" />)

    expect(
      screen.getByRole('heading', { name: /heading/i })
    ).toBeInTheDocument()

    expect(screen.getByText(/description/i)).toBeInTheDocument()
  })
})
