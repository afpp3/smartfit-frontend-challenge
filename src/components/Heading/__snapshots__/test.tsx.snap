// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render the heading 1`] = `
.c0 {
  color: #333333;
}

.c1 {
  font-weight: 800;
  font-size: 4.8rem;
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  position: relative;
  margin-bottom: 5.6rem;
}

.c1::after {
  content: '';
  position: absolute;
  bottom: -2.6rem;
  left: 0;
  width: 11.2rem;
  border-bottom: 1rem solid #333333;
}

.c2 {
  font-weight: 400;
}

@media (max-width:450px) {
  .c1 {
    font-size: 2.8rem;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    Heading
  </h1>
  <p
    class="c2"
  />
</div>
`;
